code/game/objects/structures/grille.dm

```diff
@@ -10,208 +10,215 @@
 	pressure_resistance = 5*ONE_ATMOSPHERE
 	layer = 2.9
 	explosion_resistance = 5
-	var/health = 10
-	var/destroyed = 0
+	var/health = 20 //Relatively "strong" since it's hard to dismantle via brute force
+	var/broken = 0
+
+/obj/structure/grille/examine(mob/user)
+
+	..()
+	if(!anchored)
+		user << "Its screws are loose."
+	if(broken) //We're not going to bother with the damage
+		user << "It has been completely smashed apart, only a few rods are still holding together"
 
 /obj/structure/grille/cultify()
-	var/turf/T = get_turf(src)
-	if(T)
-		T.ChangeTurf(/turf/simulated/wall/cult)
+	new /obj/structure/grille/cult(get_turf(src))
+	returnToPool(src)
 	..()
 
-/obj/structure/grille/fence/east_west
-	//width=80
-	//height=42
-	icon='icons/fence-ew.dmi'
-/obj/structure/grille/fence/north_south
-	//width=80
-	//height=42
-	icon='icons/fence-ns.dmi'
+/obj/structure/grille/proc/healthcheck(var/hitsound = 0) //Note : Doubles as the destruction proc()
+	if(hitsound)
+		playsound(loc, 'sound/effects/grillehit.ogg', 80, 1)
+	if(health <= (0.25*initial(health)) && !broken) //Modular, 1/4th of original health. Do make sure the grille isn't broken !
+		broken = 1
+		icon_state = "[initial(icon_state)]-b"
+		density = 0 //Not blocking anything anymore
+		getFromPool(/obj/item/stack/rods, get_turf(src)) //One rod set
+	if(health <= 0) //Dead
+		getFromPool(/obj/item/stack/rods, get_turf(src)) //Drop the second set of rods
+		returnToPool(src)
 
 /obj/structure/grille/ex_act(severity)
-	returnToPool(src)
+	switch(severity)
+		if(1)
+			health -= rand(30, 50)
+		if(2)
+			health -= rand(15, 30)
+		if(3)
+			health -= rand(5, 15)
+	healthcheck(hitsound = 1)
+	return
 
 /obj/structure/grille/blob_act()
-	returnToPool(src)
-
-/obj/structure/grille/meteorhit(var/obj/M)
-	returnToPool(src)
+	health -= rand(15, 30) //Grille will always be blasted, but chances of leaving things over

```

Comment:
I don't see what you mean, the damage blob and direct meteor impacts (which are obsolete) is always the same with all grille types, otherwise it wouldn't make sense